plugins/dkim_sign.js

```diff
@@ -355,7 +355,8 @@ exports.get_sender_domain = function (connection) {
         }
     }
 
-    if (!txn.header) return domain;
+    // In case of forwarding, only use the Envelope
+    if (!txn.header || txn.notes.forward) return domain;
```

Comment:
Should this be a compound conditional? Does something else set `txn.notes.forward`?

Reply:
The alias forwarding plugins sets them (at least mine does). See [here](https://github.com/X-Ryl669/haraka-alias-forward/blob/356ade4d75c575d2417cf541d4ac2766689177ae/plugins/rcpt_to.alias_forward.js#L40)

Reply:
I understand that, but why make that a compound conditional?  (those tend to increase in complexity over time.). Why not simply do this?

```js
if (txn.notes.forward) return domain; // forwarding, only use the Envelope
```

Reply:
Sure that's possible if you prefer.

Reply:
> Sure that's possible if you prefer.

Simpler is better, and that's simpler. Anyone can glance at that and know exactly when it'll execute.